source/extensions/transport_sockets/tls/ssl_socket.cc

```diff
@@ -110,7 +110,7 @@ Network::IoResult SslSocket::doRead(Buffer::Instance& read_buffer) {
   if (info_->state() != Ssl::SocketState::HandshakeComplete &&
       info_->state() != Ssl::SocketState::ShutdownSent) {
     PostIoAction action = doHandshake();
-    if (action == PostIoAction::Close || info_->state() != Ssl::SocketState::HandshakeComplete) {
+    if (action != PostIoAction::KeepOpen || info_->state() != Ssl::SocketState::HandshakeComplete) {
```

Comment:
Also change post actions in NotReadySslSocket to CloseError

Reply:
Done.